feat: support providing a cloudtrail from a different account

piotrb opened the pull request:
This is a tiny tweak which allows passing in the arn of the cloudtrail in a different account, and it will allow that cloudtrail to publish into the sns topic.

## Summary

We have a particular use case:
* we use Datadog
* we have the cloudtrail set up in the org management account
* but its logging to a S3 bucket in a separate security account as per AWS WAR guidelines.
* We can't use the S3 notification option because Datadog is using that avenue, so we wanted to use the SNS option on the cloudtrail itself.

We've been working with Clayton Sopel on your side to come up with a solution to make this work and it proved to be a very small change to the SNS topic policy, but given how this module is created, it didn't give us a nice way to slip this tweak in.

## How did you test this change?

Switched the module reference in our environment to this fork and it worked.

dmurray-lacework commented:
Make it so! 

(this commet triggers tests)

dmurray-lacework reviewed (approved):
Approved this change. But before we merge can you add an example of the scenario to examples/ folder

piotrb commented:
I added examples .. the diff on the module doc is quite large .. feels like its not been run in a little while .. so I added just the one line of diff adding the new variable :)

You may want to include a `.terraform-docs.yml` and support in-place updating of the docs (instead of asking people to paste the new docs from the command output :) )

piotrb commented:
Ok fine .. see last commit .. I added a running script, the terraform-docs config, and the fully updated readme :)

afiune commented on examples/consolidated-existing-cloudtrail-in-another-account-datadog-co-existance/security-account.tf:
```diff
@@ -0,0 +1,43 @@
+}
+
+module "aws_cloudtrail" {
+  source  = "lacework/cloudtrail/aws"
+  version = "~> 2.8"
```
```suggestion
  source  = "../../"
```

afiune replied:
We refer to the module locally instead of the registry ☝🏽

piotrb replied:
done

afiune commented:
Make it so 🎉

afiune commented on scripts/terraform-docs.sh:
```diff
@@ -0,0 +1,14 @@
+
+if which terraform-docs >/dev/null; then
```
Nice addition! I am going to ask my team to deploy this script in all our Terraform modules and also, add a new Makefile directive to run it.

<img src="https://media4.giphy.com/media/l2JIdnF6aJnAqzDgY/giphy.gif"/>

afiune reviewed (changes requested):
Please, look at this feedback and apply the diff in your PR, then we will merge and release! 

https://github.com/lacework/terraform-aws-cloudtrail/pull/147

<img src="https://media0.giphy.com/media/xT5LMHxhOfscxPfIfm/giphy.gif"/>

Note that I had to rebase from `main` since we had some CI changes, that is probably why PR passes CI. 🍏

afiune commented:
@piotrb Your work inspired us to adopt it in all our Terraform Modules! 🎉 Thank you.

We really don't want to let this PR fall behind, any chance you can take some time to
update it so we can merge and release it? 

<img src="https://media4.giphy.com/media/3oz8xZMZox78ZbWbFC/giphy.gif"/>

piotrb commented:
Alright, I rebased again :) .. applied the one tweak to the example. Was there anything else?

PengyuanZhao commented:
@piotrb we will be working on getting this merged.

PengyuanZhao commented:
@piotrb This PR is not mergable due to the unsigned commits. We can use this one instead https://github.com/lacework/terraform-aws-cloudtrail/pull/147. Could you double check and verify the changes? We can get it merged if it's good.

zekisherif commented:
Merged  #147 . Closing this out
